homeassistant/components/blue_current/button.py

```diff
@@ -0,0 +1,101 @@
+"""Support for Blue Current buttons."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from bluecurrent_api.client import Client
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Connector
+from .const import DOMAIN
+from .entity import ChargepointEntity
+
+
+@dataclass(frozen=True)
+class ChargePointButtonEntityDescriptionMixin:
+    """Mixin for the called functions."""
+
+    function: Callable[[Client, str], Coroutine[Any, Any, None]]
+
+
+@dataclass(frozen=True)
+class ChargePointButtonEntityDescription(
+    ButtonEntityDescription, ChargePointButtonEntityDescriptionMixin
+):
+    """Describes a Blue Current button entity."""
+
+
+CHARGE_POINT_BUTTONS = (
+    ChargePointButtonEntityDescription(
+        key="reset",
+        translation_key="reset",
+        function=lambda client, evse_id: client.reset(evse_id),
+        device_class=ButtonDeviceClass.RESTART,
+    ),
+    ChargePointButtonEntityDescription(
+        key="reboot",
+        translation_key="reboot",
+        function=lambda client, evse_id: client.reboot(evse_id),
+        device_class=ButtonDeviceClass.RESTART,
+    ),
+    ChargePointButtonEntityDescription(
+        key="stop_charge_session",
+        translation_key="stop_charge_session",
+        function=lambda client, evse_id: client.stop_session(evse_id),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Blue Current buttons."""
+    connector: Connector = hass.data[DOMAIN][entry.entry_id]
+    button_list: list[ButtonEntity] = [
+        ChargePointButton(
+            connector,
+            button,
+            evse_id,
+        )
+        for evse_id in connector.charge_points
+        for button in CHARGE_POINT_BUTTONS
+    ]
+
+    async_add_entities(button_list)
```

Comment:
It accepts a generator

```suggestion
    async_add_entities(
        ChargePointButton(
            connector,
            button,
            evse_id,
        )
        for evse_id in connector.charge_points
        for button in CHARGE_POINT_BUTTONS
    )
```